Modules/Core/Common/src/itkObjectFactoryBase.cxx

```diff
@@ -876,7 +876,7 @@ ObjectFactoryBase ::SynchronizeObjectFactoryBase(void * objectFactoryBasePrivate
   ObjectFactoryBasePrivate * previousObjectFactoryBasePrivate;
   previousObjectFactoryBasePrivate = GetPimplGlobalsPointer();
 
-  m_PimplGlobals = reinterpret_cast<ObjectFactoryBasePrivate *>(objectFactoryBasePrivate);
+  m_PimplGlobals = static_cast<ObjectFactoryBasePrivate *>(objectFactoryBasePrivate);
```

Comment:
Should this one be a dynamic_cast?

Reply:
Thanks for your question, HansI just double-checked: `objectFactoryBasePrivate` is really a void pointer, so it cannot by casted by `dynamic_cast`.